### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by nicolasmarino99**

### CODE REVIEW

**Comment by nicolasmarino99** [comment]
Unnecessary indentation problems

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module CountingMethods
```
Comment (maelfosso) [inline enumerable_methods.rb]: It should be `Enumerable`. Read the instructions https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#project-2-enumerable-methods

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    (0...size).my_each do |i|
+      yield self[i], i
+    end
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply enumerable_methods.rb]: Solve it please

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  def my_each_with_index
+    (0...size).my_each do |i|
+      yield self[i], i
+    end 
+  end
+
+  def my_select
+    arr = []
+    my_each do |i|
+      if yield i
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply enumerable_methods.rb]: Solve it please

File: enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        arr.push(i)
+      end
+    end
+    arr
+  end
+
+  def my_all?
+    my_each do |i|
+      unless yield i
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply enumerable_methods.rb]: Solve it please

File: enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    end
+    false   
+  end
+
+  def my_none? 
+    my_each do |i|
+      return false if yield i 
+    end
+    true   
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb:49]: Not working
```
> [nil, false, true].none?  
 => false 
> [nil, false, true].my_none?  
Traceback (most recent call last):
        7: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        6: from (irb):25
        5: from enumerable_methods.rb:43:in `my_none?'
        4: from enumerable_methods.rb:5:in `my_each'
        3: from enumerable_methods.rb:5:in `each'
        2: from enumerable_methods.rb:6:in `block in my_each'
        1: from enumerable_methods.rb:44:in `block in my_none?'
LocalJumpError (no block given (yield))
```

File: enumerable_methods.rb [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  def my_count
+    count = 0
+    my_each do |i|
+      count += 1 if yield i
+    end
+  count
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb:61]: Not working
```
> ary = [1, 2, 4, 2]
 => [1, 2, 4, 2] 
> ary.count
 => 4 
> ary.my_count
Traceback (most recent call last):
        7: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        6: from (irb):28
        5: from enumerable_methods.rb:51:in `my_count'
        4: from enumerable_methods.rb:5:in `my_each'
        3: from enumerable_methods.rb:5:in `each'
        2: from enumerable_methods.rb:6:in `block in my_each'
        1: from enumerable_methods.rb:52:in `block in my_count'
LocalJumpError (no block given (yield))
> ary.count(2)
 => 2 
> ary.my_count(2)
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):30
        1: from enumerable_methods.rb:49:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))

```

File: enumerable_methods.rb [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+  def my_inject
+    accumulator = self[0]
+    self.drop(1).my_each do |i|
+      accumulator = yield accumulator, i 
+    end
+    accumulator
+  end
+
+end
```
Comment (maelfosso) [inline enumerable_methods.rb]: Not working

```
> [5,6,7,8,9,10].inject(1) { |product, n| product * n }
 => 151200 
> [5,6,7,8,9,10].my_inject(1) { |product, n| product * n }
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):38
        1: from enumerable_methods.rb:65:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

You did a really nice job.
But before moving on, there are changes to do. 
Also, you must solve *stickler-ci* errors before moving on.

Thanks.

File: enumerable_methods.rb (hunk 7 above)
Comment (nicolasmarino99) [reply enumerable_methods.rb]: Hello, this is my display
```
`> [5,6,7,8,9,10].inject { |product, n| product * n }`
`=> 151200 `
`> [5,6,7,8,9,10].my_inject { |product, n| product * n }`
`=> 151200 `
```

**Review by bellom — APPROVED** [review]
Hi @nicolasmarino99,
Great work making all the necessary corrections from the previous review, all methods work fine now.
kindly be aware that you are to use `Git Flow` but instead you used `Github Flow`, make sure to use 'Git Flow' in the next project - `OOP`
Short Explanation Hint: 
`GitHub Flow`: Pull Requests is raise from feature branches to the master.
`Git Flow`: Pull Requests is raise from feature branches to the development.
Go ahead and merge :+1:

**Pull request merged**